packages/node-core/src/indexer/blockDispatcher/base-block-dispatcher.ts

```diff
@@ -165,13 +165,16 @@ export abstract class BaseBlockDispatcher<Q extends IQueue, DS> implements IBloc
     const {blockHash, dynamicDsCreated, reindexBlockHeight} = processBlockResponse;
 
     if (reindexBlockHeight !== null && reindexBlockHeight !== undefined) {
+      await this.storeCacheService.flushCache(true, false);
+      await this.storeCacheService.resetCache();
```

Comment:
Is there a reason for doing this here and not [here](https://github.com/subquery/subql/blob/a1ca389c828feb7ad7ead2dae386d1a4d3f2d2f2/packages/node-core/src/utils/reindex.ts#L45)?

Reply:
changed